python/main.py

```diff
@@ -1,32 +1,170 @@
 
+DB_PATH = './mercari.sqlite3'
+
+
 app = FastAPI()
 logger = logging.getLogger("uvicorn")
 logger.level = logging.INFO
 images = pathlib.Path(__file__).parent.resolve() / "image"
-origins = [ os.environ.get('FRONT_URL', 'http://localhost:3000') ]
+origins = [os.environ.get('FRONT_URL', 'http://localhost:3000')]
+
 app.add_middleware(
     CORSMiddleware,
     allow_origins=origins,
     allow_credentials=False,
-    allow_methods=["GET","POST","PUT","DELETE"],
+    allow_methods=["GET", "POST", "PUT", "DELETE"],
     allow_headers=["*"],
 )
 
+# loading inventory
+with open('inventory.json') as js:
+    iv = json.load(js)
+
+# connect to db
+conn = sqlite3.connect(DB_PATH)
+c = conn.cursor()
+
+# create table
+c.execute("""CREATE TABLE IF NOT EXISTS items (
+                id INTEGER PRIMARY KEY,
+                name STRING,
+                category STRING,
+                image STRING
+          )""")
+
+# inv = [('jacket', 'fashion'), ('cat-cage', 'pet'), ('switch', 'entertainment')]
+
+#operations on db
+# c.execute("INSERT INTO items (name,category) VALUES ('jacket', 'fashion')") #insert one
+# c.execute("INSERT INTO items (name,category_id) VALUES ('jacket', '1')") #insert one
+# c.execute('DELETE FROM category;',) #drop all data in table
+# c.execute('DELETE FROM items WHERE id = 6 or 7 or 8') # delete certain item
+# c.executemany("INSERT INTO items(name,category) VALUES(?,?)", inv) #insert many
+# c.execute("""ALTER TABLE items
+# ADD COLUMN image STRING""") # add new column
+# c.execute("""DROP table items""") #drop the table
+# RENAME COLUMN location TO image""")# rename the column
+
+# c.execute("SELECT category_id FROM items")
+# c.execute("UPDATE items SET category_id='1' WHERE category='fashion'")
+
+conn.commit()
+conn.close()
+
+
+# Hash the image
+def hash_image(filepath):
+    with open(filepath, "rb") as f:
+        bts = f.read()  # read entire file as bytes
+        readable_hash = hashlib.sha256(bts).hexdigest();
+        return (f'{readable_hash}.jpg')
+
+
+# print(hash_image('./image/default.jpg'))
+
+# functions of add/get items
+def get_all_items(json_output=False):
+    try:
+        conn = sqlite3.connect(DB_PATH)
+        c = conn.cursor()
+        sql = 'SELECT name, category, image FROM items'
+        # c.execute("""SELECT items.name,items.image,category.category_name
+        # FROM items INNER JOIN category
+        # ON items.category_id =category.category_id""")
+        c.execute(sql)
+        r = [dict((c.description[i][0], value) \
+                  for i, value in enumerate(row)) for row in c.fetchall()]
+        conn.commit()
+        conn.close()
+        return r[0] if json_output else r
+    except Error as e:
+        print(e)
+        return None
+
+
+
+def get_specific_items(**args):
+    try:
+        conn = sqlite3.connect(DB_PATH)
+        c = conn.cursor()
+        if 'name' in args: c.execute("SELECT name,category,image FROM items WHERE name=?", (args['name'].lower(),))
```

Comment:
this is fine for training, but consider https://xkcd.com/327/ for professional development